codegen/projections/high_score_service/lib/high_score_service/client.rb

```diff
@@ -63,10 +63,11 @@ def api_key_auth(params = {}, options = {})
         error_inspector_class: Hearth::HTTP::ErrorInspector
       )
       stack.use(Hearth::Middleware::Auth,
-        auth_schemes: options.fetch(:auth_schemes, config.auth_schemes),
         auth_params: Auth::Params.new(operation_name: :api_key_auth),
         http_api_key_identity_resolver: options.fetch(:http_api_key_identity_resolver, config.http_api_key_identity_resolver),
         auth_resolver: options.fetch(:auth_resolver, config.auth_resolver),
+        identity_resolver_map: { http_login_identity_resolver: Hearth::Identities::HTTPLogin, http_bearer_identity_resolver: Hearth::Identities::HTTPBearer, http_api_key_identity_resolver: Hearth::Identities::HTTPApiKey },
```

Comment:
Needing this map on every use of middleware seems wrong.  Why doesn't the auth scheme itself know the type and the symbol? 

Or why doesn't the Identity itself know the symbol?

Thinking about passing in a custom authScheme to an operation, how would this work?  It seems like the auth scheme would need to know how to get its identity if its not in that mapping?

Reply:
I think the map is always needed because it's used to find the identity resolver based on the identity, for what is only valid on the operation. We previously constructed this in middleware by collecting the identity resolver keywords. A custom auth scheme wouldn't work unless it is one of the available ones for the operation (it's a way to force a single type of auth or auth resolution chain). The auth scheme does know the type but there is no mapping from type to config, which is arbitrary.

Reply:
But the AuthScheme itself is what defines the configuration.  I think the example code has `identityResolver = authScheme.identityResolver(identityResolvers);`. It does seem like its reasonable for an authScheme to be able to say, given config, return the correct identity provider.  

But if we want to stick to auth schemes in ruby only defining their identity type and keeping this type to config map - I think it can be a constant on the generated middleware.  The type map shouldn't be specific to the operation (because config is typed at the client level.  The type of a certain config property cant change in different operations, nor can we add new ones in an operation).